pkgs/top-level/stage.nix

```diff
@@ -189,8 +189,39 @@ let
     in
     res;
 
+  mkOptionalOverlay =
+    {
+      configOption,
+      type,
+      overlay,
+    }:
+    self: super:
+    let
+      pkgs = overlay self super;
+    in
+    if config."${configOption}" then
+      pkgs
+    else
+      lib.mapAttrs (
+        name: _: throw "${name} ${type} is not available, please enable config.${configOption}"
+      ) pkgs;
```

Comment:
This complicated function is still here and is now only used once.

Reply:
Ok, removed